File: mobile/backend/app/models/order.py
"""
Order model for warehouse commands.
"""

from datetime import datetime
from typing import Optional
from pydantic import Field

from app.models.base import BaseModel
from app.core.enums import OrderType, OrderStatus


class Order(BaseModel):
    """Represents a warehouse order."""

    type: OrderType = Field(..., description="Order type")
    status: OrderStatus = Field(default=OrderStatus.PENDING, description="Order status")
    supervisor_id: Optional[str] = Field(default=None, description="Supervisor who created the order")
    validator_id: Optional[str] = Field(default=None, description="Supervisor who validated")
    validated_at: Optional[datetime] = Field(default=None, description="Validation timestamp")
    product_id: Optional[str] = Field(default=None, description="Product ID")
    quantity: int = Field(default=0, ge=0, description="Ordered quantity")

    def to_firestore(self) -> dict:
        """Convert to Firestore-compatible dict."""
        data = super().to_firestore()
        if self.validated_at:
            data["validated_at"] = self.validated_at.isoformat()
        return data

File: mobile/frontend/lib/models/order.dart
class Order {
  final String id;
  final String type;
  final String status;
  final String? supervisorId;
  final String? validatorId;
  final String? validatedAt;
  final String? productId;
  final int quantity;
  final String? createdAt;
  final String? updatedAt;

  Order({
    required this.id,
    required this.type,
    this.status = 'pending',
    this.supervisorId,
    this.validatorId,
    this.validatedAt,
    this.productId,
    this.quantity = 0,
    this.createdAt,
    this.updatedAt,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['id'] ?? '',
      type: json['type'] ?? 'command',
      status: json['status'] ?? 'pending',
      supervisorId: json['supervisor_id'],
      validatorId: json['validator_id'],
      validatedAt: json['validated_at'],
      productId: json['product_id'],
      quantity: json['quantity'] ?? 0,
      createdAt: json['created_at'],
      updatedAt: json['updated_at'],
    );
  }

  Map<String, dynamic> toJson() => {
    'type': type,
    'product_id': productId,
    'quantity': quantity,
  };

  bool get isPending => status == 'pending';
  bool get isValidated => status == 'validated';
}
